src/main/java/com/pingcap/tikv/types/IntegerType.java

```diff
@@ -110,15 +197,15 @@ public static void writeUVarLong(CodecDataOutput cdo, long value) {
      * @exception InvalidCodecFormatException wrong format of binary encoding encountered
      */
     public static long readLongFully(CodecDataInput cdi) {
-        byte flag = cdi.readByte();
+        int flag = cdi.readByte();
 
         switch (flag) {
             case INT_FLAG:
                 return readLong(cdi);
             case VARINT_FLAG:
                 return readVarLong(cdi);
             default:
-                throw new InvalidCodecFormatException("Invalid Flag type for signed long type: " + flag);
+                throw new InvalidCodecFormatException("Invalid Flag type for signed long type: " + codecMap.get(flag));
```

Comment:
please revert to original logic since when throw, it's likely flag is not valid and will cause another throw.